### PR TITLE
Projectkk2glider/issue 4138 and other fixes

### LINKED ISSUE #4138: Name/Parameter missmatch on alerts - horus
When switching to the external antenna on horus, you get  prompt which tells you to continue.. or exit.

The radio has no button.  Exit is the RTN button.

I suspect this occurs in multiple places, but to avoid confusion the names should be matched to the radios labels.

### PR DESCRIPTION
**Pull request by projectkk2glider**
Closes #4138.

### CODE REVIEW

File: radio/src/datastructs.h [hunk 1]
```diff
@@ -980,7 +980,11 @@ PACK(struct RadioData {
   NOBACKUP(int8_t backlightMode);
   NOBACKUP(TrainerData trainer);
   NOBACKUP(uint8_t view);            // index of view in main screen
-  NOBACKUP(int8_t buzzerMode:2);    // -2=quiet, -1=only alarms, 0=no keys, 1=all
+#if defined(BUZZER)
+  NOBACKUP(int8_t buzzerMode:2);    // -2=quiet, -1=only alarms, 0=no keys, 1=all (only used on AVR radios without audio hardware)
```
Comment (kilrah) [inline radio/src/datastructs.h]: Hmm these are actually completely redundant and could be ditched for all radios....
There's even some dead code in the only place it's used...

```
#if defined(BUZZER) // AUDIO + BUZZER
      case ITEM_SETUP_BUZZER_MODE:
        g_eeGeneral.buzzerMode = editChoice(RADIO_SETUP_2ND_COLUMN, y, STR_BUZZER, STR_VBEEPMODE, 

[...]

#elif defined(BUZZER) // BUZZER only
      case ITEM_SETUP_BUZZER_MODE:
        g_eeGeneral.beepMode = editChoice(RADIO_SETUP_2ND_COLUMN, y, STR_SPEAKER, STR_VBEEPMODE, g_eeGeneral.beepMode, -2, 1, attr, event);
```

Comment (projectkk2glider) [inline radio/src/datastructs.h]: I think it is still used for some AVR radios
```
if(AUDIO)
  add_definitions(-DAUDIO)
  set(SRC ${SRC} audio_avr.cpp)
  set(BUZZER OFF)
else()
  set(BUZZER ON)
endif()
if(BUZZER)
  add_definitions(-DBUZZER)
  set(SRC ${SRC} buzzer.cpp)
endif()
```

Comment (kilrah) [inline radio/src/datastructs.h]: The definitions are different, but they basically lead to the same thing being done in the code.

Comment (projectkk2glider) [inline radio/src/datastructs.h]: Your first comment is wrong, the code you pasted is not complete. First `#if` has effectively a condition `#if defined(AUDIO) && defined(BUZZER)`, the `#elif` is only executed if `AUDIO` is not defined and `BUZZER` is defined.

The only cleanup possible is macros in `adudio_arm.h` where `BUZZER` is used, but is never defined.

**Comment by bsongis** [comment]
Merge done, please check!

**Comment by fred99** [comment]
Hello,
the  following modification in datastructs.h

```
#if defined(BUZZER)
  NOBACKUP(int8_t buzzerMode:2);    // -2=quiet, -1=only alarms, 0=no keys, 1=all (only used on AVR radios without audio hardware)
#else
  NOBACKUP(int8_t spareRadio:2);
#endif
```

produces 11549 errors in MSVC compiler ! :

```
c:\users\devfr\desktop\depot-opentx\next\radio\src\datastructs.h(1022): error C2121: '#': invalid character: possibly the result of a macro expansion (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
16>c:\users\devfr\desktop\depot-opentx\next\radio\src\datastructs.h(1022): error C2059: syntax error: 'if' (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
16>c:\users\devfr\desktop\depot-opentx\next\radio\src\datastructs.h(1022): error C2334: unexpected token(s) preceding ':'; skipping apparent function body (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
16>c:\users\devfr\desktop\depot-opentx\next\radio\src\myeeprom.h(637): error C2071: 'RadioData::g_eeGeneral': illegal storage class (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
16>c:\users\devfr\desktop\depot-opentx\next\radio\src\myeeprom.h(637): error C2079: 'RadioData::g_eeGeneral' uses undefined struct 'RadioData' (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
16>c:\users\devfr\desktop\depot-opentx\next\radio\src\myeeprom.h(638): error C2071: 'RadioData::g_model': illegal storage class (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
16>C:\Users\devfr\Desktop\depot-opentx\next\radio\src\opentx.h(353): error C2071: 'RadioData::g_tmr10ms': illegal storage class (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\targets\taranis\adc_driver.cpp)
....
....
8>c:\users\devfr\desktop\depot-opentx\next\radio\src\opentx.h(490): error C2229: struct 'RadioData' has an illegal zero-sized array (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\vario.cpp)
8>c:\users\devfr\desktop\depot-opentx\next\radio\src\opentx.h(500): error C2720: 'RadioData::channel_order': 'extern' storage-class specifier illegal on members (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\vario.cpp)
8>c:\users\devfr\desktop\depot-opentx\next\radio\src\opentx.h(550): error C2071: 'RadioData::heartbeat': illegal storage class (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\vario.cpp)
8>c:\users\devfr\desktop\depot-opentx\next\radio\src\opentx.h(550): error C2229: struct 'RadioData' has an illegal zero-sized array (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\vario.cpp)
8>c:\users\devfr\desktop\depot-opentx\next\radio\src\opentx.h(550): fatal error C1003: error count exceeds 100; stopping compilation (compiling source file C:\Users\devfr\Desktop\depot-opentx\next\radio\src\vario.cpp)
========== Build: 8 succeeded, 1 failed, 10 up-to-date, 0 skipped ==========
```

If I comment to come back to previous datastructs.h file:

```
//#if defined(BUZZER)
  NOBACKUP(int8_t buzzerMode:2);    // -2=quiet, -1=only alarms, 0=no keys, 1=all (only used on AVR radios without audio hardware)
//#else
//  NOBACKUP(int8_t spareRadio:2);
//#endif
```

No more error:
```
....
19>  CMake does not need to re-run because C:\Users\devfr\Desktop\depot-opentx\next\build\CMakeFiles\generate.stamp is up-to-date.
========== Build: 19 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

**Comment by kilrah** [comment]
Ah yes... @projectkk2glider you can't have #defines in a struct on MSVC

**Comment by projectkk2glider** [comment]
659f9e05564fcc10d4f6c42711cf017c19cb17fd should fix it.

**Comment by fred99** [comment]
It's fixed, thank  
========== Build: 9 succeeded, 0 failed, 10 up-to-date, 0 skipped ==========

**Pull request merged**